File: tests/pywin/Pythonwin/pywin_/framework/window.py
# Framework Window classes.

# Most Pythonwin windows should use these classes rather than
# the raw MFC ones if they want Pythonwin specific functionality.
import win32com_.gen_py.mfc.window
import win32con


class MDIChildWnd(win32com_.gen_py.mfc.window.MDIChildWnd):
    def AutoRestore(self):
        "If the window is minimised or maximised, restore it."
        p = self.GetWindowPlacement()
        if p[1] == win32con.SW_MINIMIZE or p[1] == win32con.SW_SHOWMINIMIZED:
            self.SetWindowPlacement(p[0], win32con.SW_RESTORE, p[2], p[3], p[4])

File: tests/pywin-py2many/Pythonwin/pywin_/framework/window.jl
import win32com_.gen_py.mfc.window
import win32con
abstract type AbstractMDIChildWnd <: Abstractwin32com_.gen_py.mfc.window.MDIChildWnd end
mutable struct MDIChildWnd <: AbstractMDIChildWnd

end
function AutoRestore(self::MDIChildWnd)
#= If the window is minimised or maximised, restore it. =#
p = GetWindowPlacement(self)
if p[2] == win32con.SW_MINIMIZE || p[2] == win32con.SW_SHOWMINIMIZED
SetWindowPlacement(self, p[1], win32con.SW_RESTORE, p[3], p[4], p[5])
end
end
